openapi_python_client/templates/endpoint_module.py.jinja

```diff
@@ -40,10 +44,12 @@ def _get_kwargs(
     files = {}
     data = {}
     for key, value in multipart_data.to_dict().items():
-         if isinstance(value, File):
+         if is_file(value):
              files[key] = value
-         else:
+         elif isinstance(value, str):
              data[key] = value
+         else:
+             data[key] = json.dumps(value)
```

Comment:
Another note is that we don't do this at all for normal forms (see a few lines down), and I assume that we want to be consistent with our form handling with or without files.

Reply:
By normal forms I assume you mean `application/x-www-form-urlencoded` - maybe we should be doing this? See [relevant section in the OpenAPI spec](https://swagger.io/specification/#support-for-x-www-form-urlencoded-request-bodies)

Reply:
As my understanding, multipart form content using httpx must always be send using the `files` attrs (even if its not a real file):
https://www.python-httpx.org/advanced/#multipart-file-encoding

Sending JSON as multipart-content, is as following: 
```
client.post('http://172.21.115.183:7777/nnrf-nfm/v1', files={'upload-file': (None, '{"foo":"bar"}', 'application/json')}))
```
![image](https://user-images.githubusercontent.com/18233250/118524008-008c9280-b73e-11eb-99df-8fbc7368d964.png)

If we use the data attribute, we are not sending multipart-content:
```
client.post('http://172.21.115.183:7777/nnrf-nfm/v1', data={'upload-file': '{"foo":"bar"}'}))
```
![image](https://user-images.githubusercontent.com/18233250/118524326-53fee080-b73e-11eb-8698-543edd9b914c.png)

Reply:
As we can of course, mix multipart content-type: 
```
client.post('http://172.21.115.183:7777/nnrf-nfm/v1', 
  files={
     'upload-file': (None, '{"foo":"bar"}', 'application/json'), 
     'sometext': (None, "Foo.Bar", "text/plain")
     }
)
```

![image](https://user-images.githubusercontent.com/18233250/118525259-439b3580-b73f-11eb-84af-ea413d4f1e2a.png)

Reply:
Correction: when the `files` attr is set, it does include the `data` values inside the multipart-content

```
client.post('http://172.21.115.183:7777/nnrf-nfm/v1', files={'upload-file': (None, '{"foo":"bar"}', 'application/json'), 'sometext': (None
one, "Foo.Bar", "text/plain")}, data={'baz': 'foobar'})
```

But it definitly does not apply the specs default content-type `text/plain` for primitive: https://swagger.io/docs/specification/describing-request-body/multipart-requests/
![image](https://user-images.githubusercontent.com/18233250/118526745-d8526300-b740-11eb-9da6-0c98a510f7e7.png)

It does not specify wich encoding is used, I assumed that it should be interpreted as `text/plain` as default, wich match the default content-type define in the specs ( https://swagger.io/docs/specification/describing-request-body/multipart-requests/ )

It's looks good to me (best will be later to add support of default content-type override via the `Encoding Object
`)

Regarding the `application/x-www-form-urlencoded`; since they aren't multipart content, to me it is expected to handles them in a different way.